runtime/src/main/cpp/CppSupport.hpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+#ifndef RUNTIME_CPP_SUPPORT_H
+#define RUNTIME_CPP_SUPPORT_H
+
+#include <type_traits>
+#include <memory>
+
+// A collection of backported utilities from future C++ versions.
+
+namespace kotlin {
+
+////////////////////////// C++14 //////////////////////////
+
+template <typename T, typename... Args>
+std::unique_ptr<T> make_unique(Args... args) {
```

Comment:
I guess this does not work for move-only parameter type. Should it be `make_unique(Args&&... args)` and then `forward` ?

Reply:
Right you are, fixed.